mesonbuild/modules/pkgconfig.py

```diff
@@ -307,7 +307,7 @@ def format_reqs(self, reqs: T.List[str]) -> str:
         for name in reqs:
             vreqs = self.version_reqs.get(name, None)
             if vreqs:
-                result += [name + ' ' + self.format_vreq(vreq) for vreq in vreqs]
+                result += [name + ' ' + self.format_vreq(vreq) for vreq in sorted(vreqs)]
```

Comment:
I'm not sure sorting is the right thing there, shouldn't we preserve the order in which user has put requires?

Reply:
I believe these concerns [are addressed in this comment](https://github.com/mesonbuild/meson/pull/12528#issuecomment-1819148406). In particular, note that this PR only determines the ordering when a range is required (and only that part; crucially, the _entire_ line is not being sorted), and that nobody can be relying on the current behaviour anyway.